process/interface.go

```diff
@@ -65,6 +65,11 @@ type SmartContractResultProcessor interface {
 type SmartContractProcessorFacade interface {
 	SmartContractProcessor
 	SmartContractResultProcessor
+
+	ArgsParser() ArgumentsParser
+	TxTypeHandler() TxTypeHandler
```

Comment:
Not sure I understand this correctly, but why do we need these? 
These are the components that each scr processor receives on constructor, uses internally and are the same for each of them.

Reply:
This also applies to `CheckSCRBeforeProcessing`. This should be a separate small component that each scr processor receives and uses internally. No need to duplicate the code inside each scr processor.

Reply:
needed an interface to pass to the sovereignSCProcessorCreator, that could take either a scprocessor, a scprocessorV2, or a scProcessorProxy. 

CheckSCRBeforeProcessing was internal and made it public to be able to use it from the new interface.

Reply:
refactored to remove the ArgsParser() and TxTypeHandler()

Reply:
removed also CheckSCRBeforeProcessing